src/binary/in_memory.rs

```diff
@@ -0,0 +1,84 @@
+use crate::binary::Node;
+use crate::common::Bytes32;
+use crate::{binary, common};
+
+type StorageMap = common::StorageMap<u64, Node>;
+type BinaryMerkleTree = binary::MerkleTree<StorageMap>;
+
+pub struct MerkleTree {
+    tree: BinaryMerkleTree,
+}
+
+impl MerkleTree {
+    pub fn new() -> Self {
+        Self {
+            tree: BinaryMerkleTree::new(StorageMap::new()),
+        }
+    }
+
+    pub fn push(&mut self, data: &[u8]) {
+        let _ = self.tree.push(data);
+    }
+
+    pub fn root(&mut self) -> Bytes32 {
+        self.tree.root().unwrap()
+    }
+}
```

Comment:
Should we also have `prove` as it's a part of the public API for regular binary merkle trees?

Reply:
I can certainly add that in. `prove` is a fallible function because it takes in an index argument, and that index needs to be within the range of leaf indices.

Reply:
Would it make sense to convert the error into an Option<T> if that's the only expected failure case?

Reply:
Yeah, that simplifies the interface a bit 👍

Reply:
I think if it's obvious why the return value is None, or we have some docs, that is good. It's more idiomatic in Rust to use option when there's only one possible error / reason imo.